headjack/agents/plot_data.py

```diff
@@ -64,25 +84,30 @@ async def plot_data_agent(question: Utterance, n: int = 1, temp: float = 0.0) ->
         ret.source = "plot_data"
         return ret
     cols = cast(List[PlotDataColumn], cols)
-    code = await _plot_data(cols, question, n, temp)
+    code = await _plot_data(cols, question, n, temp, chat_context)
     if isinstance(code, Response):
         code.parent = question
         return code
-    ret = Observation(utterance=plot_json(cols, code), parent=question)
+    ret = Observation(utterance=code.utterance, metadata=plot_json(cols, code.metadata['code']), parent=question)  # type: ignore
     ret.source = "plot_data"
     return ret
 
 
-async def _plot_data(cols: List[PlotDataColumn], question: Utterance, n: int, temp: float) -> Union[str, Response]:
+async def _plot_data(
+    cols: List[PlotDataColumn],
+    question: Utterance,
+    n: int,
+    temp: float,
+    chat_context: bool,
+) -> Union[Answer, Response]:
     response = await consolidate_responses(await _plot_data_prompt(cols, question, n, temp))  # type: ignore
```

Comment:
This call to `_plot_data_prompt` missing the last arg for `chat_context: bool`. I'm guessing it should be `True`?